csharp/extractor/Semmle.Extraction.CSharp.Standalone/BuildAnalysis.cs

```diff
@@ -329,6 +331,76 @@ private void Restore(IEnumerable<string> targets)
             }
         }
 
+        private void DownloadMissingPackages(IEnumerable<string> restoreTargets)
+        {
+            var alreadyDownloadedPackages = Directory.GetDirectories(packageDirectory.DirInfo.FullName).Select(d => Path.GetFileName(d).ToLowerInvariant()).ToHashSet();
+            var notYetDownloadedPackages = new HashSet<string>();
+
+            var allFiles = GetFiles("*.*").ToArray();
```

Comment:
Shouldn't this be limited somehow? Now we are reading in every line of every file in the repo?

Reply:
I don't know what would be a good filter here. I've seen package references in `.csproj`, `.proj`, `.props` files and maybe more. On the powershell repo, the reading and regex matching takes less than 2 seconds, a single `dotnet restore` call is probably longer, as that needs to access the network.